Artifacts/MsSql/Data/Security/1060-RemoveRedundantBootstrapPermissions.sql

```diff
@@ -0,0 +1,45 @@
+-- SPDX-License-Identifier: Apache-2.0
+-- Licensed to the Ed-Fi Alliance under one or more agreements.
+-- The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+-- See the LICENSE and NOTICES files in the project root for more information.
+
+DECLARE @claim_Name NVARCHAR(MAX) = 'http://ed-fi.org/ods/identity/claims/domains/educationOrganizations';
+DECLARE @educationOrganizationsResourceClaimId INT;
+DECLARE @claimSet_Id INT;
+DECLARE @claimSet_Name NVARCHAR(MAX) = 'Bootstrap Descriptors and EdOrgs';
+DECLARE @msg NVARCHAR(MAX);
+IF NOT EXISTS (SELECT 1 FROM dbo.ResourceClaims WHERE ClaimName = @claim_Name)    
+BEGIN
+	SET @msg = 'ClaimName does not exist';
+    THROW 50000, @msg, 1
+END
+
+IF NOT EXISTS (SELECT 1 FROM dbo.ClaimSets WHERE ClaimSetName = @claimSet_Name)    
+BEGIN
+   SET @msg = 'ClaimSetName does not exist';
+   THROW 50000, @msg, 1
+END
+
+IF  EXISTS (SELECT 1 FROM dbo.ResourceClaims WHERE ClaimName = @claim_Name)    
+BEGIN
+	SELECT	@educationOrganizationsResourceClaimId = ResourceClaimId
+	FROM	dbo.ResourceClaims rc
+	WHERE	rc.ClaimName = @claim_Name
+END
+
+IF  EXISTS (SELECT 1 FROM dbo.ClaimSets WHERE ClaimSetName = @claimSet_Name)    
+BEGIN
+	SELECT	@claimSet_Id = claimSetId
+	FROM	dbo.ClaimSets rc
+	WHERE	rc.ClaimSetName = @claimSet_Name
+END
+
+
+DELETE FROM dbo.ClaimSetResourceClaims 
+WHERE  ClaimSetResourceClaimId IN (SELECT ClaimSetResourceClaimId
+FROM   dbo.ClaimSetResourceClaims csrc
+WHERE  csrc.ClaimSet_ClaimSetId = @claimSet_Id
+AND csrc.ResourceClaim_ResourceClaimId IN 
+(SELECT  ResourceClaimId
+FROM	dbo.ResourceClaims rc
+WHERE	rc.ParentResourceClaimId = @educationOrganizationsResourceClaimId))
```

Comment:
I think we should be more specific so that we don't do anything that could inadvertently impact unrelated claim assignments that might be present that we aren't targeting.

I think you should add criteria to this query to specifically target only the records in _dbo.ClaimSetResourceClaims_ which match the following criteria:

- The action (_Action_ActionId_) is Create. Initialize a `@createActionId` variable containing the _ActionId_ from the _dbo.Actions_ table and add that to the WHERE clause.
- The authorization strategy override (_AuthorizationStrategyOverride_AuthorizationStrategyId_) is for "No Further Authorization Required". Initialize a variable  containing the appropriate _AuthorizationStrategyId_ from the _dbo.AuthorizationStrategies_ table and add that to the WHERE clause.

Apply this change to the Postgres version of the script as well.

Reply:
I updated the logic as per your review comments